document/appendix/implementation.rst

```diff
@@ -0,0 +1,130 @@
+.. _impl:
+.. index:: ! implementation limitations, implementation
+
+Implementation Limitations
+--------------------------
+
+Implementations typically impose additional restrictions on a number of aspects of a WebAssembly module or execution.
+These may stem from:
+
+* physical resource limits,
+* constraints imposed by the embedder or its environment,
+* limitations of selected implementation strategies.
+
+This section lists allowed limitations.
+Where restrictions take the form of numeric limits, no minimum requirements are given,
+nor are the limits assumed to be concrete, fixed numbers.
+However, it is expected that all implementations have "reasonably" large limits  to enable common applications.
+
+.. note::
+   A conforming implementation is not allowed to leave out individual *features*.
+   However, designated subsets of WebAssembly may be specified in the future.
+
+
+Syntactic Limits
+~~~~~~~~~~~~~~~~
+
+.. _impl-syntax:
+.. index:: abstract syntax, module, type, function, table, memory, global, element, data, import, export, parameter, result, local, structured control instruction, instruction, name, Unicode, code point
+
+Structure
+.........
+
+An implementation may impose restrictions on the following dimensions of a module:
+
+* the number of :ref:`types <syntax-type>` in a :ref:`module <syntax-module>`
+* the number of :ref:`functions <syntax-func>` in a :ref:`module <syntax-module>`, including imports
+* the number of :ref:`tables <syntax-table>` in a :ref:`module <syntax-module>`, including imports
+* the number of :ref:`memories <syntax-mem>` in a :ref:`module <syntax-module>`, including imports
+* the number of :ref:`globals <syntax-global>` in a :ref:`module <syntax-module>`, including imports
+* the number of :ref:`element segments <syntax-elem>` in a :ref:`module <syntax-module>`
+* the number of :ref:`data segments <syntax-data>` in a :ref:`module <syntax-module>`
+* the number of :ref:`imports <syntax-import>` to a :ref:`module <syntax-module>`
+* the number of :ref:`exports <syntax-export>` form a :ref:`module <syntax-module>`
+* the number of parameters in a :ref:`function type <syntax-functype>`
+* the number of results in a :ref:`function type <syntax-functype>`
+* the number of :ref:`locals <syntax-local>` in a :ref:`function <syntax-func>`
+* the size of a :ref:`function <syntax-func>` body
+* the size of a :ref:`structured control instruction <syntax-instr-control>`
+* the number of :ref:`structured control instructions <syntax-instr-control>` in a :ref:`function <syntax-func>`
+* the nesting depth of :ref:`structured control instructions <syntax-instr-control>`
+* the length of an :ref:`element segment <syntax-elem>`
+* the length of a :ref:`data segment <syntax-data>`
+* the length of a :ref:`name <syntax-name>`
+* the range of :ref:`code points <syntax-codepoint>` in a :ref:`name <syntax-name>`
```

Comment:
It was a comment on an issue, with no effect on the PR's content.

I think you misunderstand what I'm trying to get at: within the standards process, we can revisit this issue as it stands. By separating it, and polling the group on it, I first expect the outcome to stay as Andreas proposes and I expect we will not revisit the issue for lack of new information (this is a requirement!).

Without separating out the issue we're sneaking it in here among otherwise unsurprising things. That's not the way I want to conduct this effort. I want to call out what *could* be contentious so that people who care clearly see it and can respond. Doing so further forces us to document our thought process.

So, I maintain that I'd like this to be separate.